File: 94. Binary Tree Inorder Traversal/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func inorderTraversal(root *TreeNode) []int {
    if root != nil {
        output := inorderTraversal(root.Left)
        output = append(output, root.Val)
        output = append(output, inorderTraversal(root.Right)...)
        return output
    } else {
        return []int{}
    }
}

File: 94. Binary Tree Inorder Traversal/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        Stack<TreeNode> stack = new Stack<TreeNode>();
        TreeNode node = root;
        List<Integer> sol = new ArrayList<Integer>();
        
        while (node != null || stack.isEmpty() == false) {
            if (node != null) {
                stack.add(node);
                node = node.left;
            } else {
                node = stack.pop();
                sol.add(node.val);
                node = node.right;
            }
        }
        
        return sol;
    }
}
